addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -28,3 +28,17 @@ def _get_proxy_identification(self, company, proxy_type):
                 raise UserError(_('Please fill your codice fiscale to be able to receive invoices from FatturaPA'))
             return company.partner_id._l10n_it_edi_normalized_codice_fiscale()
         return super()._get_proxy_identification(company, proxy_type)
+
+    def _toggle_proxy_user_active(self):
+        """
+        Toggle the value of the ``active`` boolean field of the proxy_user,
+        and handle sending the reactivate/deactivate requests to the IAP side.
+        """
+        self.ensure_one()
+        self.active = not self.active
+        server_url = self._get_proxy_urls()['l10n_it_edi'][self.edi_mode]
+
+        if self.active:
+            self._make_request(f"{server_url}/api/l10n_it_edi/1/reactivate_user")
+        else:
+            self._make_request(f"{server_url}/api/l10n_it_edi/1/deactivate_user")
```

Comment:
This works but is a bit counter-intuitive IMO, since you just toggled `active` :sweat_smile: 
Plus, if the `_make_request()` fails for any reason, you might end up with an inconsistent state between Odoo and IAP.